### PR TITLE
Turned Pytorch losses into `torch.nn.module` classes

### PR DESCRIPTION
**Pull request by kdgutier**
Changed pytorch losses from normal classes to `torch.nn.module`, with parameters without gradient. This allows Pytorch lightning to automatically infer the gpu device where the loss parameters should be stored during training and inference.

This solves the uncertainty intervals example notebook breaking in Google Colab.

### CODE REVIEW

**Review by mergenthaler — CHANGES_REQUESTED** [review]
Please include one sentence in the documentation explaining that we are inheriting `torch.nn.module` from Lighting to infer GPU devices. It might help users if something breaks.

**Review by kdgutier — COMMENTED** [review]
You need to approve the changes

**Review by mergenthaler — APPROVED** [review]
Thanks for the changes.
